substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/SubstrateGraphBuilderPlugins.java

```diff
@@ -399,6 +409,65 @@ private static void interceptUpdaterInvoke(MetaAccessProvider metaAccess, Snippe
         }
     }
 
+    private static void registerServiceLoaderPlugins(SnippetReflectionProvider snippetReflection, InvocationPlugins plugins, ImageClassLoader loader, boolean analysis) {
+        Registration serviceLoaderRegistration = new Registration(plugins, ServiceLoader.class);
+        serviceLoaderRegistration.register1("load", Class.class, new InvocationPlugin() {
```

Comment:
You also need to register a plugin for `public static <S> ServiceLoader<S> load(Class<S> service, ClassLoader loader)`. You can ignore the `loader` argument, just like we to for [`ReflectionPlugins`](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/ReflectionPlugins.java#L118) since the `ImageClassLoader` is used to load the resources anyway.